contracts/ServiceNodeContribution.sol

```diff
@@ -86,6 +91,11 @@ contract ServiceNodeContribution is Shared {
         operator = tx.origin; // NOTE: Creation is delegated by operator through factory
         blsPubkey = _blsPubkey;
         serviceNodeParams = _serviceNodeParams;
+
+        for (uint256 i = 0; i < _reservedContributors.length; i++) {
+            reservedContributorAddresses.push(_reservedContributors[i].addr);
+            reservedContributions[_reservedContributors[i].addr] += _reservedContributors[i].stakedAmount;
+        }
```

Comment:
This needs to test as it goes that the reserved contribution amounts are meeting the minimum contribution requirement, given the operator stake + any previous reserved contributions added during the loop.    That is, `[25% operator, 10%, 10%, 15%, 50%]` and `[25% operator, 70%, 5%]` are acceptable reservations, but `[25% operator, 10%, 5%]` is not.

But that said, see my other comment about the operator contribution not being known at this point.

(There are also some slightly weird cases where `[25% operator, 70%, 5%]` is an acceptable reservation, but `[25%, 5%, 70%]` is not, but we have those limitations in current OXEN reservations as well, so I don't think it's necessary to worry about them in the contract; the front-end/contract caller can just sort them in descending order first if it wants to allow those sorts of things.)